src/py21cmfast/drivers/single_field.py

```diff
@@ -215,10 +222,13 @@ def compute_halo_grid(
     ----------
     initial_conditions : :class:`~InitialConditions`
         The initial conditions of the run.
-    perturbed_halo_list: :class:`~PerturbHaloField` or None, optional
+    inputs : :class:`~InputParameters`
+        The input parameters specifying the run. Since this will be the first box
+        to use the astro params/flags, if it is computed this is always needed.
```

Comment:
This is not how I intended it -- the astro params/flags are gotten from the initial conditions if they're not given explicitly here, because in almost every case you are setting this up via `run_coeval` or `run_lightcone` where the full inputs are defined before anything is run.

Reply:
Fixed, see above